### PR TITLE
Temporarily force deletion of pods when replacing nodes

### LINKED ISSUE #730: longevity-scylla-operator-3h-gke-terminate-replace: drived scylla to have broken command line
TestId: `801e256a-2bca-4c72-be78-40711c7838c9`
Job:
`https://jenkins.scylladb.com/view/scylla-operator/job/scylla-operator/job/operator-1.4/job/gke/job/longevity-scylla-operator-3h-gke-terminate-replace/3`

After running nemesis pod `sct-cluster-us-east1-b-us-east1-2` failed to not recover

Logs showed the following:
```
command used: "/usr/bin/scylla --log-to-syslog 0 --log-to-stdout 1 --default-log-level info --network-stack posix --io-properties-file=/etc/scylla.d/io_properties.yaml --cpuset 0-7 --smp 7 --overprovisioned --listen-address 0.0.0.0 --rpc-address 0.0.0.0 --seed-provider-parameters seeds=10.83.242.82 --broadcast-address 10.83.240.56 --broadcast-rpc-address 10.83.240.56 --replace-address-first-boot --blocked-reactor-notify-ms 999999999"
parsed command line options: [log-to-syslog: 0, log-to-stdout: 1, default-log-level: info, network-stack: posix, io-properties-file: /etc/scylla.d/io_properties.yaml, cpuset: 0-7, smp: 7, overprovisioned, listen-address: 0.0.0.0, rpc-address: 0.0.0.0, seed-provider-parameters: seeds=10.83.242.82, broadcast-address: 10.83.240.56, broadcast-rpc-address: 10.83.240.56, replace-address-first-boot: --blocked-reactor-notify-ms, (positional) 999999999]
error: too many positional options have been specified on the command line
```

As you can see `--replace-address-first-boot` had no value: `--replace-address-first-boot --blocked-reactor-notify-ms 999999999`

Probably we need to guard to agains empty value on sidecar side
https://github.com/scylladb/scylla-operator/blob/be0e67fb21da920f07e4107334bf71af7e367940/pkg/sidecar/config/config.go#L230-L232

**To Reproduce**
1. Deploy scylla-operator on gke
2. Pick scylla pod to attack
3. Find node where pod is hosted
4. Drain k8s node: `kubectl drain node <node>` 
5. Uncordon k8s node: `kubectl uncordon node <node>` 
6. Wait till pod get scheduled again
7. Mark pod to be replaced: `kubectl label svc <svc> scylla/replace=""'`
8. Wait till pod is recreated and get to up and running state
 
**Expected behavior**
Pod is successfully recovered

**Logs**
db-cluster  | https://cloudius-jenkins-test.s3.amazonaws.com/801e256a-2bca-4c72-be78-40711c7838c9/20210724_185703/db-cluster-801e256a.tar.gz
monitor-set | https://cloudius-jenkins-test.s3.amazonaws.com/801e256a-2bca-4c72-be78-40711c7838c9/20210724_185703/monitor-set-801e256a.tar.gz
loader-set  | https://cloudius-jenkins-test.s3.amazonaws.com/801e256a-2bca-4c72-be78-40711c7838c9/20210724_185703/loader-set-801e256a.tar.gz
sct-runner  | https://cloudius-jenkins-test.s3.amazonaws.com/801e256a-2bca-4c72-be78-40711c7838c9/20210724_185703/sct-runner-801e256a.tar.gz
kubernetes  | https://cloudius-jenkins-test.s3.amazonaws.com/801e256a-2bca-4c72-be78-40711c7838c9/20210724_185703/kubernetes-801e256a.tar.gz 

**Environment:**
 - Platform: GKE
 - Kubernetes version: 
 - Scylla version: 4.4.2
 - Scylla-operator version: v1.4.0-beta.0-0-gf382529

### PR DESCRIPTION
**Pull request by tnozicka**
Because of a Kubernetes bug, when evicting unready pods https://github.com/kubernetes/kubernetes/issues/103970 we have to temporarily use Delete calls and get back to Evict when the kube bug gets fixed in all supported version. The revert has a tracking issue #732.

**Which issue is resolved by this Pull Request:**
Resolves #730

### CODE REVIEW

File: pkg/controller/scyllacluster/sync_services.go [hunk 1]
```diff
@@ -244,18 +243,28 @@ func (scc *Controller) syncServices(
...
 				"Pod", klog.KObj(podMeta),
 			)
-			err = scc.kubeClient.CoreV1().Pods(podMeta.Namespace).Evict(ctx, &policyv1beta1.Eviction{
-				ObjectMeta: metav1.ObjectMeta{
-					Name: podMeta.Name,
-				},
-				DeleteOptions: &metav1.DeleteOptions{
-					PropagationPolicy: &backgroundPropagationPolicy,
-				},
+			/*
```
Comment (zimnx) [inline pkg/controller/scyllacluster/sync_services.go:256]: Not a fan of commented code, TODO comment about coming back to eviction is enough

**Comment by tnozicka** [comment]
hm, this is actually fixed in 1.20

File: pkg/controller/scyllacluster/sync_services.go (hunk 1 above)
Comment (tnozicka) [reply pkg/controller/scyllacluster/sync_services.go]: ok

**Comment by tnozicka** [comment]
we can't wait for kube 1.19.z (first chance is [11. 8.](https://kubernetes.io/releases/patch-releases/#1-19)) so let's merge this and backport to 1.4 for now

**Review by zimnx — APPROVED** [review]

**Pull request merged**